centml/cli/cluster.py

```diff
@@ -40,10 +40,18 @@ def get_hw_to_id_map():
 # # Hardware pricing tier that loads choices dynamically
 class HardwarePricingTier(click.ParamType):
     def __init__(self):
-        self.hw_to_id_map, self.id_to_hw_map = get_hw_to_id_map()
-        self.choices = list(self.hw_to_id_map.keys())
+        self.hw_to_id_map = None
+        self.id_to_hw_map = None
+        self.choices = None
+
+    def initialize_maps(self):
+        if self.hw_to_id_map is None or self.id_to_hw_map is None or self.choices is None:
+            self.hw_to_id_map, self.id_to_hw_map = get_hw_to_id_map()
+            self.choices = list(self.hw_to_id_map.keys())
 
     def convert(self, value, param, ctx):
+        # calling initialize_maps to defer api call during initialization phase
+        self.initialize_maps()
```

Comment:
is this for not making the hardware API call if the user command doesn't require hw details?

Reply:
so after making the global instance variable for handling the api call(if you recall the last changes which we have recently done), during the initialization phase of the code, its still making the api call to fetch hardware instances even if we write centml to the cli command after logout which is resulting into the failure of the api call.

In order to avoid making the api call we have deferred the api call to the convert method which is eventually trigger when we fill the hardware choices and it will only make the api call then

Reply:
yes, just to avoid the api call if its not required to execute

Reply:
Created an issue to write unit tests for CLI as well: https://github.com/CentML/centml-python-client/issues/63

This would allow us to catch all these issues.

Reply:
sure @anandj91